File: 2188-minimized-maximum-of-products-distributed-to-any-store/minimized-maximum-of-products-distributed-to-any-store.py
class Solution:
    def minimizedMaximum(self, n: int, quantities: List[int]) -> int:
        l = 1
        r = max(quantities)
        result = 0
        while l <= r:
            m = l + (r - l) // 2
            s = 0
            for i in quantities:
                s += ceil(i / m)
            if s <= n:
                r = m - 1
                result = m
            else:
                l = m + 1
        return result

File: 2188-minimized-maximum-of-products-distributed-to-any-store/minimized-maximum-of-products-distributed-to-any-store.cpp
class Solution {
public:
    int minimizedMaximum(int n, vector<int>& quantities) {
        int l = 1;
        int r = *max_element(quantities.begin(), quantities.end());
        int result = 0;
        while (l <= r){
            int m = l + int((r - l) / 2);
            int s = 0;
            for(float q: quantities)
                s += ceil(q/m);
            if (s <= n){
                r = m - 1;
                result = m;
            }
            else l = m + 1;
        }
        return result;
    }
};